src/main/java/bio/terra/cli/utils/WorkspaceManagerUtils.java

```diff
@@ -98,30 +103,38 @@ public static WorkspaceDescription createWorkspace(ApiClient apiClient) {
 
       // create the Google project that backs the Terra workspace object
       UUID jobId = UUID.randomUUID();
-      CreateGoogleContextRequestBody contextRequestBody = new CreateGoogleContextRequestBody();
-      contextRequestBody.setJobId(jobId.toString());
-      workspaceApi.createGoogleContext(contextRequestBody, workspaceId);
+      CreateCloudContextRequest cloudContextRequest = new CreateCloudContextRequest();
+      cloudContextRequest.setCloudContext(CloudContext.GOOGLE);
+      cloudContextRequest.setJobControl(new JobControl().id(jobId.toString()));
+      workspaceApi.createCloudContext(cloudContextRequest, workspaceId);
 
-      // poll the get workspace endpoint until the project id property is populated
+      // poll the job result endpoint until the job status is completed
       final int MAX_JOB_POLLING_TRIES = 120; // maximum 120 seconds sleep
       int numJobPollingTries = 1;
-      String googleProjectId;
+      JobReport jobReport;
       do {
-        workspaceWithContext = workspaceApi.getWorkspace(workspaceId);
-        googleProjectId =
-            (workspaceWithContext.getGoogleContext() == null)
-                ? null
-                : workspaceWithContext.getGoogleContext().getProjectId();
         logger.info(
-            "job polling try #{}, workspace context: {}, project id: {}",
+            "job polling try #{}, workspace id: {}, job id: {}",
             numJobPollingTries,
-            workspaceWithContext.getId(),
-            googleProjectId);
+            workspaceId,
+            jobId);
+        jobReport = jobsApi.retrieveJob(jobId.toString());
+        logger.debug("create workspace jobReport: {}", jobReport);
         numJobPollingTries++;
-        if (googleProjectId == null) {
+        if (jobReport.getStatus().equals(JobReport.StatusEnum.RUNNING)) {
```

Comment:
With the new API you could instead poll on `workspaceApi.createCloudContextResult`, which has a `jobReport` property. 
The biggest difference would be that you get an ErrorReport if something goes wrong during creation, though I don't know whether you want to surface that for the CLI or not. It also provides the actual result, although that doesn't matter since you're returning a whole WorkspaceDescription and not just the context.

Reply:
Good point about the ErrorReport. I added it to the debug logging.